src/plots/cartesian/axes.js

```diff
@@ -1604,7 +1604,7 @@ axes.doTicks = function(gd, axid, skipTitle) {
     var axletter = axid.charAt(0),
         counterLetter = axes.counterLetter(axid),
         vals = axes.calcTicks(ax),
-        datafn = function(d) { return d.text + d.x + ax.mirror; },
+        datafn = function(d) { return [d.text, d.x, ax.mirror].join('\u0000'); },
```

Comment:
Actually, perhaps a character that's not possible in a stringified number, like `'_'`? It would be rather pathological, but Imagine you had
```js
xaxis: {
    type: 'linear',
    title: 'Grade',
    tickvals: [-1, 1],
    ticktext: ['A', 'A-']
}
```

(Nice debugging BTW @rpaskowitz !)